examples/5-steps-with-keras.ipynb

```diff
@@ -0,0 +1,522 @@
+    "                      loss='sparse_categorical_crossentropy',\n",
+    "                      metrics=['accuracy', 'sparse_categorical_crossentropy'])"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "The step with Keras model needs both train and validation data. So far we took care of reshaping only the train data. The step below fills this hole."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "reshape_valid_step = Step(\n",
+    "    name=\"reshape_valid\",\n",
+    "    transformer=ReshapeData(),\n",
+    "    input_data=['input_valid'],\n",
+    "    adapter={\n",
+    "        'X': [('input_valid', 'images')],\n",
+    "        'y': [('input_valid', 'labels')]\n",
+    "    },\n",
+    "    cache_dirpath=CACHE_DIR\n",
+    ")"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "Now we have all dependencies necessary to add the crucial step."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "cnn_step = Step(\n",
+    "    name=\"CNN\",\n",
+    "    transformer=KerasCnn(),\n",
+    "    input_steps=[loader_step, reshape_step, reshape_valid_step],\n",
+    "    adapter={\n",
+    "        'datagen': [(loader_step.name, 'datagen')],\n",
+    "        'X_train': [(reshape_step.name, 'X')],\n",
+    "        'y_train': [(reshape_step.name, 'y')],\n",
+    "        'X_valid': [(reshape_valid_step.name, 'X')],\n",
+    "        'y_valid': [(reshape_valid_step.name, 'y')]\n",
+    "    },\n",
+    "    cache_dirpath=CACHE_DIR\n",
+    ")"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "cnn_step"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "The pipeline looks fine. Time to run it!"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {
+    "scrolled": true
+   },
+   "outputs": [],
+   "source": [
+    "result = cnn_step.fit_transform(data)"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "def eval_pred(title, y_true, y_pred):\n",
+    "    print(title)\n",
+    "    print(\"  Log-loss: \", log_loss(y_true=y_true, y_pred=y_pred))\n",
+    "    choices = np.argmax(y_pred, axis=1)\n",
+    "    print(\"  Accuracy: {:.2%}\".format(np.sum(choices == y_true) / len(y_true)))\n",
+    "    \n",
+    "eval_pred(\"Results on training\", y_true=y_train, y_pred=result['output'])\n",
+    "eval_pred(\"Results on validation\", y_true=y_valid, y_pred=result['output_valid'])"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "Because we do test time augmentation, it makes sense to run prediction phase a few times and average the results.\n",
```

Comment:
@grzes314 It wasn't explained anywhere before (I think) that we are augmenting at inference time and hence it actually is tta.